ShellPkg/Application/Shell/ShellManParser.c

```diff
@@ -633,13 +634,19 @@ ProcessManFile (
       FileDevPath = FileDevicePath (NULL, TempString);
       // MU_CHANGE [START] - CodeQL change
       if (FileDevPath == NULL) {
+        Status = EFI_OUT_OF_RESOURCES;
+        goto Done;
+      }
+
+      DevPath = AppendDevicePath (ShellInfoObject.ImageDevPath, FileDevPath);
+
+      if (DevPath == NULL) {
         Status = EFI_INVALID_PARAMETER;

```

Comment:
I think setting `Status` to `EFI_OUT_OF_RESOURCES` made sense if `FileDevPath` is NULL. I'd expect the same status (`EFI_OUT_OF_RESOURCES`) to be returned if the append operation fails (`DevPath` is `NULL`).

Is there a specific reason you chose `EFI_INVALID_PARAMETER`?

Reply:
Out of resources wasn't listed in the function header and I didn't dive too far into the call stack. Out of resources is fine.